### ISSUE TITLE
option to report config warnings as an error / exit with a non-zero code

### ISSUE THREAD

**Issue by DetachHead** [rvben/rumdl#726]
### Problem or motivation

currently, it seems like config related issues are always reported as a warning and don't impact the exit code:

```
[inline config warning] /path/to/file.md:262: Unknown rule in inline disable-line comment: asdf
```

### Proposed solution

an option to report these as an error / exit with a non-zero code.

ideally these errors would also be present in the language server, like normal rules.

### Feature category

Other

### Example markdown

```markdown

```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature

**Comment by rvben** [comment]
The CLI half of this shipped in v0.2.35: the new `--deny-config-warnings` flag treats configuration warnings as errors (exit code 2), and it explicitly covers your example - an unknown rule name in an inline `disable-line` comment (96d75f56).

The second half of your request - surfacing these as diagnostics in the language server - isn't implemented yet. I'll leave this open to track the LSP piece. Let me know if that part still matters to you.

**Comment by DetachHead** [comment]
thanks! the language server part arguably isn't as important, as long as invalid config is blocked from getting merged by the CI, but it's still nice to have

**Comment by rvben** [comment]
The language server half shipped in v0.2.48. Inline configuration problems are now published as diagnostics under the code `inline-config`, so your example arrives in the editor rather than only in a log:

```
[inline-config] line 3 col 1: Unknown rule in inline disable-line comment: asdf
```

Configuration *file* problems deliberately stay in the server log. They belong to no document, so there is no line to attach a diagnostic to. On the CLI those are covered by `--deny-config-warnings` (v0.2.35), which is the CI gate you said matters most.

Thanks @DetachHead for the follow-up that ranked the two halves; it is why the CLI side went first.
